perceval/backends/mozilla/crates.py

```diff
@@ -0,0 +1,344 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2017 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA.
+#
+# Authors:
+#     Valerio Cosentino <valcos@bitergia.com>
+#
+
+import json
+import logging
+import requests
+import time
+
+from grimoirelab.toolkit.datetime import (datetime_utcnow,
+                                          datetime_to_utc,
+                                          str_to_datetime)
+from grimoirelab.toolkit.uris import urijoin
+
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser,
+                        metadata)
+from ...errors import CacheError
+from ...utils import DEFAULT_DATETIME
+
+
+CRATES_URL = "https://crates.io/"
+CRATES_API_URL = 'https://crates.io/api/v1/'
+
+SLEEP_TIME = 300
+
+logger = logging.getLogger(__name__)
+
+
+class Crates(Backend):
+    """Crates.io backend for Perceval.
+
+    This class allows the fetch the packages stored in Crates.io
+
+    :param sleep_time: sleep time in case of connection lost
+    """
+    version = '0.1.0'
+
+    def __init__(self, sleep_time=SLEEP_TIME, tag=None, cache=None):
+
+        origin = CRATES_URL
+
+        super().__init__(origin, tag=tag, cache=cache)
+        self.client = CratesClient(sleep_time=sleep_time)
+
+    @metadata
+    def fetch(self, category="crates", from_date=DEFAULT_DATETIME):
+        """Fetch package data.
+
+        The method retrieves packages and summary from Crates.io.
+
+        :param from_date: obtain packages updated since this date
+
+        :returns: a summary and crate items
+        """
+
+        if category == "summary":
+            return self.__fetch_summary()
+        else:
+            return self.__fetch_crates(from_date)
+
+    @metadata
+    def fetch_from_cache(self):
```

Comment:
As the this method is not implemented, don't redefine it. Parent class will throw the right exception.